### PR TITLE
Add folder_prefix option

### LINKED ISSUE #50: Improve workflow for papers
The experience of working on a paper with a large `vp_runcards` folder from where figures are produced could be better when it comes to naming files.  At the moment we have something like

```
.
├── paper.tex
├── plots
│   └── sec_somename_plot_pdfs_g.pdf
└── vp_runcards
    ├── output
    │   └── figures
    │       └── plot_pdfs_g.pdf
    └── plot_some_pdfs.yaml

```

where typically `output` is not commited to the repository and instead the results are copied and renamed manually to the `plots` directory. The problem with that approach is that redoing the plots (for example to improve the style) is quite a chore, as one needs to identify the relevant files in the paper, the find the runcard that produces them, make the changes, run validphys, hope that `output` is clean and doesn't lead to some kind of confussion, and rename the files manually as appropriate back (which may or may not be a task involving regular expressions). This is error prone and somewhat time consuming when one has to do it many times.

An easy solution would be to change the way we work to e.g. subfolders from each invocation of vp and commit the results of running it. The problem with this is that arxiv doesn't like subfolders, so instead the filename of the plots should have enough uniqueness. For that I think life would be simpler if we had a flag like `--folder-prefix` (or better name) that

  - Had as default name for the output folder to be the stem of the runcard filename.
  - Prepended the runcard filename as prefix to all figures and tables.

Then if we had something like say `pdfs_40vs31.yaml` we would get `pdfs_40vs31-plot_pdfs_g.pdf`, which is an acceptable enough name to refer to in the tex, avoids conflicts with other plot_pdfs_g from different runcards and can be easy copy pasted (or preferably simlinked).

cc @enocera

### PR DESCRIPTION
**Pull request by Zaharid**
Add an option to add a prefix to figures and tables that are named
automatically.

I decided against modifying the default output  folder name because the
interaction with the existing -o option is a bit  weird, and it is easy
enough to pass it.

Closes #50.

### CODE REVIEW

File: src/reportengine/app.py [hunk 1]
```diff
@@ -192,6 +192,12 @@ def argparser(self):
         parallel.add_argument('--no-parallel', dest='parallel',
                               action='store_false')
 
+        parser.add_argument(
+            '--folder-prefix',
+            action='store_true',
+            help="add a prefix derived from the runcard name to ourput resources.",
```
Comment (siranipour) [inline src/reportengine/app.py:198]:
```suggestion
            help="add a prefix derived from the runcard name to output resources.",
```

File: src/reportengine/environment.py [hunk 2]
```diff
@@ -43,6 +44,11 @@ def __init__(self, *, output=None, formats=('pdf',),
         self.loglevel = loglevel
         self.extra_args = kwargs
         self.config_yml = config_yml
+        if folder_prefix and config_yml:
```
Comment (siranipour) [inline src/reportengine/environment.py:47]: Out of interest when will config_yml be false? When using the API?

File: src/reportengine/environment.py [hunk 3]
```diff
@@ -116,7 +122,7 @@ def ns_dump_description(cls):
         return dict(
             output_path = "Folder where the results are to be written.",
             config_rel_path = cls.config_rel_path.__doc__,
-
+            filename_prefix = "Prefix for to prepend to filenames",
```
Comment (siranipour) [inline src/reportengine/environment.py:125]:
```suggestion
            filename_prefix = "Prefix prepended to filenames",
```

File: src/reportengine/formattingtools.py [hunk 4]
```diff
@@ -27,6 +27,9 @@ def get_nice_name(ns, nsspec, suffix=None):
     (see the ``namespaces`` documentation for more details)"""
     parts = []
     currspec = []
+    prefix = ns.get("filename_prefix", None)
```
Comment (siranipour) [inline src/reportengine/formattingtools.py:30]: Is the `filename_prefix` resolved from `ns_dump_description` or by checking the attributes of `environment`? Presumably the environment object is transformed to a dictionary at some point right?

**Review by siranipour — COMMENTED** [review]
Seems to work for me

File: src/reportengine/environment.py (hunk 2 above)
Comment (Zaharid) [reply src/reportengine/environment.py:47]: Yes, for example.

File: src/reportengine/formattingtools.py (hunk 4 above)
Comment (Zaharid) [reply src/reportengine/formattingtools.py:30]: Yes. Not proud of that interface. See `ns_dump` below.

**Pull request merged**